source/code/providers/support/mycnf-scanner.cpp

```diff
@@ -0,0 +1,244 @@
+/*
+ * --------------------------------- START OF LICENSE ----------------------------
+ *
+ * MySQL cimprov ver. 1.0
+ *
+ * Copyright (c) Microsoft Corporation
+ *
+ * All rights reserved. 
+ *
+ * MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the ""Software""), to deal
+ * in the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+ * of the Software, and to permit persons to whom the Software is furnished to do
+ * so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ * ---------------------------------- END OF LICENSE -----------------------------
+*/
+/**
+    \file        mycnf-scanner.cpp
+
+    \brief       Scans mysql configuration to extract authentication information.
+
+    \date        2014-11-05 16:22:00
+
+*/
+/*----------------------------------------------------------------------------*/
+
+#include "mycnf-scanner.h"
+
+namespace MySQL
+{
+    static inline bool GetPathFromDirective(const std::wstring& line, SCXCoreLib::SCXFilePath& path)
+    {
+        std::vector<std::wstring> elements;
+        SCXCoreLib::StrTokenize(line, elements, L" ");
+        if ( elements.size() != 2 )
+        {
+            return false;
+        }
+
+        path.Set(elements[1]);
+        return true;
+    }
+
+    bool MySQL_ConfigurationScanner::LoadCnfFile(
+        const SCXCoreLib::SCXFilePath& file, 
+        MySQL_Authentication& auth,
+        std::set< std::wstring >& visited, 
+        std::map<unsigned int, std::wstring>& ports)
+    {
+        SCX_LOGHYSTERICAL(m_log, SCXCoreLib::StrAppend(L" loading: ", file.Get()));
+        std::pair<std::set< std::wstring >::iterator, bool> ins = visited.insert(file.Get());
+        if (!ins.second)
+        {
+            std::wstringstream ss;
+            ss << L"Already visited " << file.Get() << L". skipping";
+            SCX_LOGWARNING(m_log, ss.str());
```

Comment:
I'm not that far yet. But it seems like this is written in a way to not allow testing this capability. I guess I'll see that when I get to the unit test.